src/Agoda.IoC.NetCore/StartupExtension.cs

```diff
@@ -49,7 +49,9 @@ public static IServiceCollection AutoWireAssembly<T>(this IServiceCollection ser
                         var factoryInstance = Activator.CreateInstance(reg.FactoryType);
                         var buildMethod = factoryInstance.GetType().GetMethod("Build");
                         Debug.Assert(buildMethod != null, nameof(buildMethod) + " != null"); // type is checked by RegistrationInfo.Validate()
-                        return buildMethod.Invoke(factoryInstance, null);
+
+                        var componentResolverInstance = Activator.CreateInstance(typeof(NetCoreComponentResolver), x);
```

Comment:
If you know the type during compile-time then why not just use `new` construction instead of `activator.CreateInstance`?